charts/hono/templates/_helpers.tpl

```diff
@@ -482,10 +497,10 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- .Values.prometheus.server.fullnameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
 {{- $name := default "prometheus" .Values.prometheus.nameOverride -}}
-{{- if contains $name .Release.Name -}}
-{{- printf "%s-%s" .Release.Name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
+{{- if contains $name ( include "hono.fullname" . ) -}}
+{{- printf "%s-%s" ( include "hono.fullname" . ) .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
-{{- printf "%s-%s-%s" .Release.Name $name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
+{{- printf "%s-%s-%s" ( include "hono.fullname" . ) $name .Values.prometheus.server.name | trunc 63 | trimSuffix "-" -}}
```

Comment:
The Prometheus service resource is using a name created in the context of the Prometheus chart.
Therefore, using the Hono fullname isn't correct here.
The changes in these lines should just be reverted, using `.Release.Name` again instead of `( include "hono.fullname" . )`.